### PR TITLE
Simplify early examples and update README for NAN 2

### PR DESCRIPTION
**Pull request by dfabulich**

### CODE REVIEW

**Comment by toszter** [comment]
Is there anyone with write access that will accept all these constructive pull requests? Hopefully not asleep at the wheel. Much of the nan examples cause warnings because they haven't been updated since Node v0.12 (yikes!)

**Comment by mhdawson** [comment]
@dfabulich this repo has been re-animated with a focus on N-API examples.  Do you believe this change is still needed/appropriate?

**Comment by dfabulich** [comment]
Great to hear that this repo is returning to life. My PR is about NAN, of course. I assume that the repo will continue to include NAN documentation for the historical record? If so, I think my PR changes do make sense to apply to the NAN directory, even if the primary focus of the repository switches to N-API.

**Comment by mhdawson** [comment]
@kkoopa any chance you could review this as well?

File: 3_callbacks/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+```c++
+#include <nan.h>
+
+NAN_METHOD(RunCallback) {
+  v8::Local<v8::Function> cb = info[0].As<v8::Function>();
+  const unsigned argc = 1;
+  v8::Local<v8::Value> argv[argc] = { Nan::New("hello world").ToLocalChecked() };
+  Nan::MakeCallback(Nan::GetCurrentContext()->Global(), cb, argc, argv);
```
Comment (kkoopa) [inline 3_callbacks/README.md]: This should use [`Nan::Call`](https://github.com/nodejs/nan/blob/master/doc/maybe_types.md#api_nan_call). `Nan::MakeCallback` should only be used when originating from libuv. They will behave the same in most cases, but there are some fine details in certain corner cases that can lead to unwanted behavior.

Comment (dfabulich) [reply 3_callbacks/README.md]: I was just documenting what's already there in the example, which I didn't modify in this case. https://github.com/nodejs/node-addon-examples/blob/master/3_callbacks/nan/addon.cc

I think we should merge this as-is and file a separate issue that the example (and then its documentation) should be updated.

**Comment by mhdawson** [comment]
This is quite old, and we don't have involvement on the NAN side for the examples at this point so I'm going to close this PR. Please comment/reopen if you don't think that is the right thing to do.

**Pull request closed without merging**